Bug in solve_with_interpolation

Apparently the recent changes I made to solve_with_interpolation are having problems (possibly just because the characteristic is chosen too small in the test code), but I've seen the following twice now:

```
Generic.Mat.can_solve_with_solution_interpolation: Error During Test at /home/runner/work/AbstractAlgebra.jl/AbstractAlgebra.jl/test/generic/Matrix-test.jl:2104
  Got exception outside of a @test
  Not enough interpolation points in ring
  Stacktrace:
   [1] error(::String) at ./error.jl:33
   [2] can_solve_with_solution_interpolation_inner(::AbstractAlgebra.Generic.MatSpaceElem{AbstractAlgebra.Generic.Poly{AbstractAlgebra.Generic.Res{BigInt}}}, ::AbstractAlgebra.Generic.MatSpaceElem{AbstractAlgebra.Generic.Poly{AbstractAlgebra.Generic.Res{BigInt}}}) at /home/runner/work/AbstractAlgebra.jl/AbstractAlgebra.jl/src/generic/Matrix.jl:2438
   [3] can_solve_with_solution_interpolation(::AbstractAlgebra.Generic.MatSpaceElem{AbstractAlgebra.Generic.Poly{AbstractAlgebra.Generic.Res{BigInt}}}, ::AbstractAlgebra.Generic.MatSpaceElem{AbstractAlgebra.Generic.Poly{AbstractAlgebra.Generic.Res{BigInt}}}) at /home/runner/work/AbstractAlgebra.jl/AbstractAlgebra.jl/src/generic/Matrix.jl:2544
   [4] macro expansion at /home/runner/work/AbstractAlgebra.jl/AbstractAlgebra.jl/test/generic/Matrix-test.jl:2132 [inlined]
   [5] macro expansion at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Test/src/Test.jl:1083 [inlined]
   [6] top-level scope at /home/runner/work/AbstractAlgebra.jl/AbstractAlgebra.jl/test/generic/Matrix-test.jl:2105
   [7] include at ./boot.jl:317 [inlined]
   [8] include_relative(::Module, ::String) at ./loading.jl:1044
   [9] include(::Module, ::String) at ./sysimg.jl:29
   [10] include(::String) at ./client.jl:392
   [11] top-level scope at none:0
   [12] include at ./boot.jl:317 [inlined]
   [13] include_relative(::Module, ::String) at ./loading.jl:1044
   [14] include(::Module, ::String) at ./sysimg.jl:29
   [15] include(::String) at ./client.jl:392
   [16] top-level scope at none:0
   [17] include at ./boot.jl:317 [inlined]
   [18] include_relative(::Module, ::String) at ./loading.jl:1044
   [19] include(::Module, ::String) at ./sysimg.jl:29
   [20] include(::String) at ./client.jl:392
   [21] top-level scope at none:0
   [22] include at ./boot.jl:317 [inlined]
   [23] include_relative(::Module, ::String) at ./loading.jl:1044
   [24] include(::Module, ::String) at ./sysimg.jl:29
   [25] include(::String) at ./client.jl:392
   [26] top-level scope at none:0
   [27] eval(::Module, ::Any) at ./boot.jl:319
   [28] exec_options(::Base.JLOptions) at ./client.jl:243
   [29] _start() at ./client.jl:425
Test Summary:                                     | Pass  Error  Total
Generic.Mat.can_solve_with_solution_interpolation |   44      1     45
ERROR: LoadError: LoadError: LoadError: LoadError: Some tests did not pass: 44 passed, 0 failed, 1 errored, 0 broken.
in expression starting at /home/runner/work/AbstractAlgebra.jl/AbstractAlgebra.jl/test/generic/Matrix-test.jl:2104
in expression starting at /home/runner/work/AbstractAlgebra.jl/AbstractAlgebra.jl/test/Rings-test.jl:12
in expression starting at /home/runner/work/AbstractAlgebra.jl/AbstractAlgebra.jl/test/AbstractAlgebra-test.jl:4
in expression starting at /home/runner/work/AbstractAlgebra.jl/AbstractAlgebra.jl/test/runtests.jl:113
ERROR: Package AbstractAlgebra errored during testing
Stacktrace:
 [1] pkgerror(::String, ::Vararg{String,N} where N) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/Types.jl:120
 [2] #test#65(::Bool, ::Function, ::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/Operations.jl:1357
 [3] #test at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:0 [inlined]
 [4] #test#44(::Bool, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:278
 [5] #test#43 at ./none:0 [inlined]
 [6] #test at ./none:0 [inlined]
 [7] #test#40 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:257 [inlined]
 [8] (::getfield(Pkg.API, Symbol("#kw##test")))(::NamedTuple{(:coverage,),Tuple{Bool}}, ::typeof(Pkg.API.test)) at ./none:0
 [9] top-level scope at none:0
Error: Process completed with exit code 1.
```

See https://github.com/wbhart/AbstractAlgebra.jl/runs/2086717722?check_suite_focus=true

--- a/src/generic/Matrix.jl
+++ b/src/generic/Matrix.jl
@@ -2467,7 +2467,8 @@ function can_solve_with_solution_interpolation_inner(M::AbstractAlgebra.MatElem{
          if !flag
             return flag, r, p, pv, zero(x), zero(R)
          end
-         # Check that new solution has the same pivots as previous ones
+         p = inv!(p)
+	 # Check that new solution has the same pivots as previous ones
          if r != rnk || p != prm || pv != pivots
             if r < rnk # rank is too low: reject
                pt += 1

